### PR TITLE
ratelimit support prometheus

### PR DESCRIPTION
**Pull request by zirain**
Right now, envoy-gateway controller and proxy support metric in prometheus, this patch will make ratelimit as same as the other components.

### CODE REVIEW

File: internal/infrastructure/kubernetes/ratelimit/resource.go [hunk 1]
```diff
@@ -91,7 +91,9 @@ const (
 	// ReadinessPath is readiness path for readiness probe.
 	ReadinessPath = "/healthcheck"
 	// ReadinessPort is readiness port for readiness probe.
-	ReadinessPort = 8080
+	ReadinessPort  = 8080
+	StatsdPort     = 9125
```
Comment (arkodg) [inline internal/infrastructure/kubernetes/ratelimit/resource.go:95]: can we use `19001` for prom, similar to rest https://gateway.envoyproxy.io/latest/install/install-helm/#open-ports ?

Comment (zirain) [reply internal/infrastructure/kubernetes/ratelimit/resource.go:95]: make sense, will update later

**Review by arkodg — APPROVED** [review]
LGTM thanks for adding this enhancement !

**Review by Xunzhuo — APPROVED** [review]

File: api/v1alpha1/envoygateway_types.go [hunk 2]
```diff
@@ -326,6 +326,25 @@ type RateLimit struct {
...
+}
+
+type RateLimitMetrics struct {
+	// Prometheus defines the configuration for prometheus endpoint.
+	Prometheus *RateLimitMetricsPrometheusProvider `json:"prometheus,omitempty"`
+}
+
+type RateLimitMetricsPrometheusProvider struct {
+	// Disable the Prometheus endpoint.
+	Disable bool `json:"disable,omitempty"`
```
Comment (zhaohuabing) [inline api/v1alpha1/envoygateway_types.go:347]: Does this need to be this deep?

Comment (zirain) [reply api/v1alpha1/envoygateway_types.go:347]: keep same as others

**Review by arkodg — APPROVED** [review]

**Pull request merged**